tests/unit/Promise.ts

```diff
@@ -76,19 +70,33 @@ export function addPromiseTests(suite: any, Promise: PromiseType) {
 			}));
 		},
 
-		'sparse array': function () {
-			let dfd = this.async();
-			let iterable: any[] = [];
+		'sparse array': {
+			all() {
+				let dfd = this.async();
+				let iterable: any[] = [];
 
-			iterable[0] = Promise.resolve(0);
-			iterable[3] = Promise.resolve(3);
+				iterable[0] = Promise.resolve(0);
+				iterable[3] = Promise.resolve(3);
 
-			Promise.all(iterable).then(dfd.callback(function (value: number[]) {
-				assert.strictEqual(value[0], 0);
-				assert.isUndefined(value[1]);
-				assert.isUndefined(value[2]);
-				assert.strictEqual(value[3], 3);
-			}));
+				Promise.all(iterable).then(dfd.callback(function (value: number[]) {
+					assert.strictEqual(value[0], 0);
+					assert.isUndefined(value[1]);
+					assert.isUndefined(value[2]);
+					assert.strictEqual(value[3], 3);
+				}));
+			},
+
+			race() {
+				let dfd = this.async();
+				let iterable: any[] = [];
+
+				iterable[0] = Promise.resolve(0);
+				iterable[3] = Promise.resolve(3);
+
+				Promise.race(iterable).then(dfd.callback(function (value: number) {
+					assert.strictEqual(value, 0);
```

Comment:
Interesting. `Promise.race` and `Promise.all` operate on iterables, and ES6 array iterators don't do anything special for sparse arrays. In Chrome and FF, the following iterates over the undefined values as well as the explicitly set values:

```js
var q = [];
q[2] = 2;
q[4] = 4;
var i = q[Symbol.iterator]();
var n = i.next();
while (!n.done) {
    console.log(n.value);
    n = i.next();
}
```

Promise should be updated to match.